actions/tamper.py

```diff
@@ -105,28 +151,129 @@ def parse(self, string, logger):
         # Count the number of params in this given string
         num_parameters = string.count(":")
 
-        # If num_parameters is greater than 3, it's not a valid tamper action
-        if num_parameters > 3 or num_parameters < 2:
+        # If num_parameters is greater than 4, it's not a valid tamper action
+        if num_parameters > 4 or num_parameters < 2:
             msg = "Cannot parse tamper action %s" % string
             logger.error(msg)
             raise Exception(msg)
         params = string.split(":")
-        if num_parameters == 3:
-            self.tamper_proto_str, self.field, self.tamper_type, self.tamper_value = params
+        if num_parameters == 4:
+            # HTTP replace or insert
+            self.tamper_proto_str, self.field, self.tamper_type, self.start_index, self.encoded_payload = params
+            self.start_index = int(self.start_index)
+            self.tamper_proto = actions.utils.string_to_protocol(self.tamper_proto_str)
+            self.decoded_payload = bytes(urllib.parse.unquote(self.encoded_payload), "UTF-8")
+
+        elif num_parameters == 3:
+            # HTTP corrupt or delete could be here, check for those first
+            self.tamper_proto_str = params[0]
             self.tamper_proto = actions.utils.string_to_protocol(self.tamper_proto_str)
-            if "options" in self.field:
-                if not self.tamper_value:
-                    self.tamper_value = '' # An empty string instead of an empty byte literal
-
-            # tamper_value might be parsed as a string despite being an integer in most cases.
-            # Try to parse it out here
-            try:
-                if "load" not in self.field:
-                    self.tamper_value = int(self.tamper_value)
-            except:
-                pass
-        else:
+            if self.tamper_proto_str == "HTTPRequest":
```

Comment:
Does the user literlaly have to type "HTTPRequest" to use this? can we just make it "HTTP" like the other ones ?